splink/postgres/postgres_linker.py

```diff
@@ -0,0 +1,273 @@
+        self._register_extensions()
+
+        # Create splink schema
+        self._create_splink_schema()
+
+        super().__init__(
+            input_tables,
+            settings_dict,
+            accepted_df_dtypes,
+            set_up_basic_logging,
+            input_table_aliases=input_aliases,
+        )
+
+    def _table_to_splink_dataframe(self, templated_name, physical_name):
+        return PostgresDataFrame(templated_name, physical_name, self)
+
+    def _execute_sql_against_backend(self, sql, templated_name, physical_name):
+        # In the case of a table already existing in the database,
+        # execute sql is only reached if the user has explicitly turned off the cache
+        self._delete_table_from_database(physical_name)
+
+        sql = f"CREATE TABLE {physical_name} AS {sql}"
+        self._log_and_run_sql_execution(sql, templated_name, physical_name)
+
+        return self._table_to_splink_dataframe(templated_name, physical_name)
+
+    def _run_sql_execution(
+        self, final_sql: str, templated_name: str = None, physical_name: str = None
+    ):
+        with self._engine.connect() as con:
+            res = con.execute(text(final_sql))
+        return res
+
+    def _table_registration(self, input, table_name):
+        if isinstance(input, dict):
+            input = pd.DataFrame(input)
+        elif isinstance(input, list):
+            input = pd.DataFrame.from_records(input)
+
+        # Will error if an invalid data type is passed
+        input.to_sql(
+            table_name,
+            con=self._engine,
+            index=False,
+            if_exists="replace",
+            schema=self._db_schema,
+        )
+
+    def register_table(self, input, table_name, overwrite=False):
+        # If the user has provided a table name, return it as a SplinkDataframe
+        if isinstance(input, str):
+            return self._table_to_splink_dataframe(table_name, input)
+
+        # Check if table name is already in use
+        exists = self._table_exists_in_database(table_name)
+        if exists:
+            if not overwrite:
+                raise ValueError(
+                    f"Table '{table_name}' already exists in database. "
+                    "Please use the 'overwrite' argument if you wish to overwrite"
+                )
+            else:
+                self._delete_table_from_database(table_name)
+
+        self._table_registration(input, table_name)
+        return self._table_to_splink_dataframe(table_name, table_name)
+
+    def _random_sample_sql(self, proportion, sample_size, seed=None):
+        if proportion == 1.0:
+            return ""
+        if seed:
+            # TODO: we could maybe do seeds by handling it in calling function
+            # need to execute setseed() in surrounding session
+            raise NotImplementedError(
+                "Postgres does not support seeds in random "
+                "samples. Please remove the `seed` parameter."
+            )
```

Comment:
I think this should fail at the start of the `u_training` step, not here.

Do you have a view? It's probably something for a separate PR to address.

Reply:
Yeah I think that would be useful - same applies to `sqlite`. Maybe need to end up with `linker._supports_seed` or something